### PR TITLE
Refactor event handlers

### PR DESCRIPTION
**Pull request by michelfinley**
## Summary

This PR enhances and fixes the event handling as described in issue #60 .

Every class that handles events should now have a handle_event function, that should be called by the handle_event function of its parent. The root of all event handling is the event_loop function in main.py.

The handle_event function must always return a boolean value indicating whether the event could be handled and should not be handled by any other game element.

## Checklist

- [X] I have tested this change locally and it works as expected.
- [X] I have made sure that the code follows the formatting and style guidelines of the project.

## Labels
`area: code quality`, `game-ui`, `priority: medium`, `type: bugfix`, `type: enhancement`

### CODE REVIEW

File: src/overlay/soil.py [hunk 1]
```diff
@@ -25,7 +36,18 @@ def __init__(self, pos, group):
...
 class SoilLayer:
-    def __init__(self, all_sprites, tmx_map, frames, sounds):
+    all_sprites: pygame.sprite.Group
+    level_frames: dict
+
+    soil_sprites: pygame.sprite.Group
+    water_sprites: pygame.sprite.Group
+    plant_sprites: pygame.sprite.Group
+
+    map: dict[tuple[int, int], Tile]
```
Comment (fortwoone) [inline src/overlay/soil.py:46]: I'd advise against using this name, as it shadows a built-in iterator (see https://docs.python.org/3/library/functions.html#map).
You could rename it to `tile_map` to differentiate it from said iterator

**Review by fortwoone — CHANGES_REQUESTED** [review]
This could be merged as-is, but some code readability issues (thankfully pretty minor)

File: main.py [hunk 2]
```diff
@@ -63,8 +63,13 @@ def __init__(self):
         }
         self.current_state = GameState.MAIN_MENU
 
-    def switch_state(self, state):
+    def switch_state(self, state: GameState):
         self.current_state = state
+        if self.game_paused():
+            self.level.player.blocked = True
+            self.level.player.direction.update((0, 0))
```
[the remark this answers is not in the document]
Comment (michelfinley) [reply main.py:71]: I had problems with being off by a few decimal places when trying to use this in NPC pathfinding, so I got used to use pygame.Vector2 resp. pygame.rect.Rect.update instead.

File: main.py [hunk 3]
```diff
@@ -61,15 +63,20 @@ def __init__(self):
             GameState.PAUSE: self.pause_menu,
             GameState.SETTINGS: self.settings_menu,
             GameState.SHOP: self.shop_menu,
-            GameState.LEVEL: self.level
+            # GameState.LEVEL: self.level
```
Comment (fortwoone) [inline main.py:66]: What exactly does this achieve?

Comment (fortwoone) [reply main.py:66]: (Don’t pay attention to this comment, I read further in the code. My bad)

**Review by fortwoone — APPROVED** [review]
LGTM

**Pull request merged**